### ISSUE TITLE
SP5: Issue finding plot coordinates in WPF

### ISSUE THREAD

**Issue by anttikoskimaa** [ScottPlot/ScottPlot#3364]
**Issue:** I tried to implement draggable horizontal line to my SP5 WPF plot following the winforms demo https://github.com/ScottPlot/ScottPlot/blob/main/src/ScottPlot5/ScottPlot5%20Demos/ScottPlot5%20WinForms%20Demo/Demos/DraggableAxisSpans.cs but I'm unable to find the axis coordinate.

With the code below the horizontal line is found way below the actual line, i.e. the arrow turns to SizeNS below Y=80. I suspect that it's because the coordinate conversion does not take into account the padding above and/or below the plot area or maybe I'm doing something wrong since there is no example on finding the plot coordinate from mouse coordinate.

**ScottPlot Version:** https://www.nuget.org/packages/ScottPlot.WPF/5.0.21

**Code Sample:**

Plot is created in the xaml: `<ScottPlot:WpfPlot x:Name="Plot" />`

Code behind: 

```cs
public Page()
{
    Plot.Plot.PlottableList.Add(new HorizontalLine() { IsDraggable = true, Y = 80, Text = "80" });
    Plot.Plot.Axes.AutoScale();
    Plot.MouseMove += Plot_MouseMove;
}


private void Plot_MouseMove(object sender, System.Windows.Input.MouseEventArgs e)
{
    var pos = e.GetPosition(Plot);
    float x = (float)pos.X;
    float y = (float)pos.Y;
    Debug.WriteLine($"{x}, {y}");
    var rect = Plot.Plot.GetCoordinateRect(x, y, radius: 10);
    AxisLine lineUnderMouse = null;
    foreach (AxisLine axLine in Plot.Plot.GetPlottables<AxisLine>().Reverse())
    {
        if (axLine.IsUnderMouse(rect))
        {
            lineUnderMouse = axLine;
            break;
        }

    }

    // set cursor based on what's beneath the plottable
    if (lineUnderMouse is null) Cursor = Cursors.Arrow;
    else if (lineUnderMouse.IsDraggable && lineUnderMouse is VerticalLine) Cursor = Cursors.SizeWE;
    else if (lineUnderMouse.IsDraggable && lineUnderMouse is HorizontalLine) Cursor = Cursors.SizeNS;
}
```

**Comment by anttikoskimaa** [comment]
I noticed that I cannot reproduce this when I am using only the laptop display, but only when I am using external display so I looked into the display settings. I had 125% scale factor in the displays so when I set the scale factor to 100% the code above is working correctly.

**Comment by MCF** [comment]
Possibly the same problem as in https://github.com/ScottPlot/ScottPlot/issues/3327?

If yes there is an associated PR waiting to be merged that fixes the issue.

**Comment by anttikoskimaa** [comment]
I don't know. It was the display's scale factor that affected the issue and not ScottPlot's. I mean in Windows settings, System -> Display -> Scale was 125% instead of 100%.

**Comment by MCF** [comment]
Sorry, my mistake I didn't read your code thoroughly enough.  I think you are using WPF?  If that is true then you are missing a step when converting screen pixels to coordinates.  Oddly I have an open pull request documenting that step here:

https://github.com/ScottPlot/ScottPlot.NET/pull/34

**Comment by anttikoskimaa** [comment]
Yes I was using WPF. Thank you! My problem is resolved either by setting the `WpfPlot1.Plot.ScaleFactor = WpfPlot1.DisplayScale;` or translating the coordinates with `x *= WpfPlot1.DisplayScale;`. If both are used, the coordinates are again wrong. Maybe that could be explained in the PR?

This does not work if DisplayScale = Plot.ScaleFactor = 1.25 for example
```
            if (WpfPlot1.DisplayScale != 1.0)
            {
                x *= WpfPlot1.DisplayScale;
                y *= WpfPlot1.DisplayScale;
            }
```

I made a test case where the DisplayScale is 1.25 and I manually set the Plot's ScaleFactor to 1.5. Then the coordinate transalation worked with

```
            float x = (float)pos.X * WaveFormPlot.DisplayScale / WaveFormPlot.Plot.ScaleFactor;
            float y = (float)pos.Y * WaveFormPlot.DisplayScale / WaveFormPlot.Plot.ScaleFactor;
```

Or is this something that the #3327 should fix?

**Comment by MCF** [comment]
Yes you have the solution in your last code example above.  I'm afraid my documentation PR is not entirely correct.  To test out your suggestion above, and create a complete working example, I have re-implemented the ScottPlot demo for draggable axis lines - but for WPF instead of WinForms.  The re-implementation is found in the WPF sandbox on the high_dpi_tests branch of my ScottPlot fork.  The full code behind is found at this link:

https://github.com/MCF/ScottPlot/blob/78225df44a5cdb14552eb78b7a97e17e1c465c17/src/ScottPlot5/ScottPlot5%20Sandbox/Sandbox.WPF/MainWindow.xaml.cs

The most important function is my implementation of your idea above.  I created a GetScaledMousePosition function and those lines are here:

https://github.com/MCF/ScottPlot/blob/78225df44a5cdb14552eb78b7a97e17e1c465c17/src/ScottPlot5/ScottPlot5%20Sandbox/Sandbox.WPF/MainWindow.xaml.cs#L91-L104

I feel fairly confident that this sample application covers all cases of scaled plots and/or scaled displays.  But this is relatively complicated so I will have to revisit my documentation PR.

And of course #3327 is unrelated to your difficulties.

**Comment by anttikoskimaa** [comment]
Thanks a lot for the quick responses! I'll try that.

**Comment by MohammadSadeghSalimian** [comment]
I also have this issue with SP5:
`private void WpfPlot_MouseDown(object sender, MouseButtonEventArgs e)
 {
     if (e.LeftButton != MouseButtonState.Pressed || Keyboard.Modifiers != ModifierKeys.Control) return;
     
     var p = e.GetPosition(WpfPlot);
     var cc = WpfPlot.Plot.GetCoordinates((float)p.X, (float)(p.Y));
     _newPointEvent.OnNext(cc);
 }`
It gives me the wrong coordinate. My laptop screen is a 4K screen with a 250% scale.

**Comment by MCF** [comment]
If you look at the complete working example I linked in my comment from 3 days ago you should be able to fix your problem.

**Comment by MCF** [comment]
@anttikoskimaa, @MohammadSadeghSalimian there were bugs in the GetCoordinateRect methods related to ScaleFactor.  I've fixed the bugs and that PR has been merged (see link above).  Once version 5.0.22 is released you will need to change any code using GetCoordinateRect when ScaleFactor is != 1.  Essentially the code we worked out above includes a workaround for the bugs in ScottPlot.  Now you will need to remove the workaround from any of your code.

I've updated my example of the draggable axis line demo to work with the upcaoming 5.0.22 release.  The relevant changes are in here:

https://github.com/MCF/ScottPlot/blob/8839461d9450ccdbc844a1dd76b9d9067db40e44/src/ScottPlot5/ScottPlot5%20Sandbox/Sandbox.WPF/MainWindow.xaml.cs

More specifically:

https://github.com/MCF/ScottPlot/blob/8839461d9450ccdbc844a1dd76b9d9067db40e44/src/ScottPlot5/ScottPlot5%20Sandbox/Sandbox.WPF/MainWindow.xaml.cs#L91-L120
